hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/GzipCodec.java

```diff
@@ -154,7 +155,7 @@ public CompressionOutputStream createOutputStream(OutputStream out,
   public Compressor createCompressor() {
     return (ZlibFactory.isNativeZlibLoaded(conf))
       ? new GzipZlibCompressor(conf)
-      : null;
+      : new BuiltInGzipCompressor(conf);
```

Comment:
we also need to update `getCompressorType`.

Reply:
yea, will update.